Incompatible function pointer types in generated code

### Description

The C runtime uses a few incompatible function pointer types. Newer C compilers won't compile these.

### Steps to Reproduce

```bash
$ cd testsuite/simulation/modelica/tearing
$ omc dynamicTearing2.mos 
[...]
dynamicTearing2_03lsy.c:399:49: error: incompatible function pointer types assigning to 'int (*)(DATA *, threadData_t *)' (aka 'int (*)(struct DATA *, struct threadData_s *)') from 'int (DATA *)' (aka 'int (struct DATA *)') [-Wincompatible-function-pointer-types]
  399 |   linearSystemData[2].strictTearingFunctionCall = dynamicTearing2_eqFunction_39;
      |                                                 ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/clang  -Os -DOM_HAVE_PTHREADS -fPIC -DOM_HAVE_PTHREADS     -I\"/workspaces/OpenModelica/build_cmake/install_cmake/bin/../include/omc/c\" -I\"/workspaces/OpenModelica/build_cmake/install_cmake/bin/../include/omc\" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=dynamicTearing2 -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=3 -DOMC_NUM_NONLINEAR_SYSTEMS=0 -DOMC_NDELAY_EXPRESSIONS=0 -DOMC_NVAR_STRING=0  -c -o dynamicTearing2_14lnz.o dynamicTearing2_14lnz.c
1 error generated.
```

### Expected Behavior

Use a compatible function pointer.

### Version and OS

 - OpenModelica Version: v1.26.0-dev-491-gb022a1ae23-cmake
 - OS: Ubuntu 24.04.3 LTS
 - clang version 18.1.3 (1ubuntu1)

--- a/OMCompiler/Compiler/Template/CodegenC.tpl
+++ b/OMCompiler/Compiler/Template/CodegenC.tpl
@@ -2268,7 +2268,7 @@ template functionInitialLinearSystems(list<SimEqSystem> linearSystems, String mo
   "Generates functions in simulation file."
 ::=
   let &tempeqns = buffer ""
-  let &tempeqns += (linearSystems |> eq as SES_LINEAR(alternativeTearing = SOME(__)) => 'int <%symbolName(modelNamePrefix,"eqFunction")%>_<%equationIndex(eq)%>(DATA*);' ; separator = "\n")
+  let &tempeqns += (linearSystems |> eq as SES_LINEAR(alternativeTearing = SOME(__)) => 'int <%symbolName(modelNamePrefix,"eqFunction")%>_<%equationIndex(eq)%>(DATA* data, threadData_t* threadData);' ; separator = "\n")
   let &globalConstraintsFunctions = buffer ""
   let linearbody = functionInitialLinearSystemsTemp(linearSystems, modelNamePrefix, &globalConstraintsFunctions)
   <<
@@ -2661,7 +2661,7 @@ template functionInitialNonLinearSystems(list<SimEqSystem> nonlinearSystems, Str
   "Generates functions in simulation file."
 ::=
   let &tempeqns = buffer ""
-  let &tempeqns += (nonlinearSystems |> eq as SES_NONLINEAR(alternativeTearing = SOME(__)) => 'int <%symbolName(modelNamePrefix,"eqFunction")%>_<%equationIndex(eq)%>(DATA*, threadData_t*);' ; separator = "\n")
+  let &tempeqns += (nonlinearSystems |> eq as SES_NONLINEAR(alternativeTearing = SOME(__)) => 'int <%symbolName(modelNamePrefix,"eqFunction")%>_<%equationIndex(eq)%>(DATA* data, threadData_t* threadData);' ; separator = "\n")
   let &globalConstraintsFunctions = buffer ""
   let nlsbody = functionInitialNonLinearSystemsTemp(nonlinearSystems, modelNamePrefix, &globalConstraintsFunctions)
   <<

